src/lists.py

```diff
@@ -101,8 +118,7 @@ def keep(x: DLList[T], p: Callable[[T], bool]) -> None:
     >>> print(x)
     [2, 4]
     """
-    ...
-
+    return DLList(el for el in x.into_generator() if p(el))
```

Comment:
The specification says that the function returns `None`. You are supposed to change the list in-place, not create a new list. The two operations are different; if you have an application with several pointers to a data structure, it can matter immensely if you are updating the structure they all have references to, or you are creating a new structure that they might never know about.